modules/core-components/src/main/java/org/mule/runtime/core/internal/source/scheduler/SchedulerFlowProcessingTemplate.java

```diff
@@ -46,6 +48,10 @@ public void sendResponseToClient(CoreEvent response, Map<String, Object> paramet
   @Override
   public void sendFailureResponseToClient(MessagingException exception, Map<String, Object> parameters,
                                           CompletableCallback<Void> callback) {
+    if (defaultSchedulerMessageSource.featureFlaggingService != null
```

Comment:
pass the flow as parameter to the constructor instead of setting the visibility to protected.